doc/docs/modules/ROOT/pages/writing.adoc

```diff
@@ -286,9 +286,9 @@ The generated query will be
 
 [source,cypher]
 ----
-UNNIND $events AS event
-MERGE (n:`Person`:`Customer` {name: event.keys.name, surname: event.keys.surname})
-SET n += event.properties
+UNWIND $events AS event
+MERGE (n:`Person`:`Customer` {name: event.name, surname: event.surname})
```

Comment:
why you removed the `keys` property?

Reply:
> why you removed the `keys` property?

Because there's no `keys` property. The DataFrame we are writing has Person objects. A Person has `name` and `surname`, but no `keys`.

(I mean I could be wrong. I'm using this API for the first time.)

Reply:
Hi @darabos , thanks for your PR. 

Regarding the `keys` property: I would say it's needed. The `$events` parameter is generated [here](https://github.com/neo4j-contrib/neo4j-spark-connector/blob/4.0/src/main/scala/org/neo4j/spark/service/MappingService.scala#L23-L42), in particular look at lines 29, where we set the `keys` property.

In addition, I just added a PR to the Spark Notebooks repo https://github.com/utnaf/spark-connector-notebooks/pull/4 with the merge example, so you can try it out if you want. 
If you enable Spark logs on Zeppelin logs you will see this:

```
Writing a batch of 3 elements to Neo4j,
for jobId=60b6dced-d8aa-4d0d-b823-cf6784675403 and partitionId=0
with query: UNWIND $events AS event
MERGE (node:Person:Musician {id: event.keys.id})
SET node += event.properties
```

I guess we can put the `keys` back =) Thank you!

Reply:
> Regarding the `keys` property: I would say it's needed. The `$events` parameter is generated [here](https://github.com/neo4j-contrib/neo4j-spark-connector/blob/4.0/src/main/scala/org/neo4j/spark/service/MappingService.scala#L23-L42), in particular look at lines 29, where we set the `keys` property.

I see! So when we use a custom query, `keys` and `properties` are not there, but they are present when we let the connector generate the query.

> I guess we can put the `keys` back =) Thank you!

Done. I've put back `properties` too. Sorry for the delay.